### PR TITLE
Select which calendars are displayed (#69)

### LINKED ISSUE #69: Select which calendars are displayed
**Description**
Currently, when allowing mDone to display calendar entries, it shows all entries from all calendars. I have lots of calendars, often shared with family members / co-workers. It would be great to be able to disable certain calendars from showing in mDone so we only see the entries that are relevant within the context of mDone taks.

### PR DESCRIPTION
**Pull request by marco308**
## Summary

Adds **Settings → Calendar → "Calendars in mDone"**, letting users choose which device calendars contribute events to mDone's Calendar and Today views. Previously mDone showed events from *every* calendar, so users with shared family/work calendars had no way to keep that noise out of their task context.

- Toggle list of the device's event calendars (with colour dots) + **Show All / Hide All**.
- Selections persist via a UserDefaults-backed `HiddenCalendarStore`.
- The store persists the **hidden** set (not the visible set), so the empty default means "show everything" and any newly added calendar appears automatically until explicitly hidden. Stale ids for deleted calendars are pruned.
- Filtering logic lives entirely in the pure, injectable `HiddenCalendarStore` rather than the EventKit-backed `CalendarService`, so it is fully unit-testable.
- `AppState` exposes an observable `calendarFilterToken`; the Calendar screen keys an event refetch on it, so toggling a calendar refreshes the month grid and selected-day list immediately — not just the Today view.

### Files

| File | Change |
|---|---|
| `mDone/App/CalendarSelection.swift` *(new)* | `CalendarInfo` + `HiddenCalendarStore` (pure filter, prune, persistence) |
| `mDone/App/AppState.swift` | `availableCalendars()`, observable `calendarFilterToken`, `calendarSelectionDidChange()` |
| `mDone/Services/CalendarService.swift` | `availableCalendars()`; filtering applied in `fetchEvents` (day/month inherit it) |
| `mDone/Models/CalendarEvent.swift` | Added `calendarIdentifier`; EventKit-free memberwise init for tests |
| `mDone/Views/Settings/CalendarSelectionView.swift` *(new)* | The toggle UI |
| `mDone/Views/Settings/SettingsScreen.swift` | New "Calendar" section linking to it |
| `mDone/Views/Calendar/CalendarScreen.swift` | Refetch keyed on `calendarFilterToken` |
| Tests + `CHANGELOG.md` | See below |

## Related Issues

Fixes #69

## Testing

- New `mDoneTests/CalendarSelectionTests.swift` — 15 tests covering the store (default/empty semantics, set/unset, idempotency, cross-instance persistence, replace, prune, multi-calendar filtering, empty-identifier edge case) and `CalendarInfo`/`CalendarEvent` equality. Uses an isolated `UserDefaults` suite so `.standard` is never polluted.
- `mDoneTests/AppStateTests.swift` — +2 tests for `calendarFilterToken` bumping.
- Full unit suite green: **264/264** (the only failure in a full run is the pre-existing, unrelated `mDoneUITests.testAppLaunchesToSetupScreen`).
- Coverage: the new `CalendarSelection.swift` is at **91.5%** (above the 85% service target; the uncovered lines are the `EKCalendar` initialiser, which can't be unit-tested without a live EventKit store). `CalendarService`'s EventKit methods remain untestable in-harness by design — which is exactly why the filter logic was factored into the pure store.
- Verified end-to-end on the iOS Simulator: hiding UK Holidays removed its bank-holiday events from both the month-grid dots and the day list; re-enabling it and navigating straight back to the Calendar tab (without changing the date) brought them back immediately, confirming the live-refresh.

## Checklist

- [x] Builds and runs on iOS (verified on iPhone 17 simulator). macOS not separately built, but no iOS-only APIs were introduced — the new core/service/model code is platform-neutral and the one `#if os(iOS)` guard (list style) follows existing patterns.
- [x] Tests pass (264 unit tests; new tests included)
- [x] `swiftlint lint --quiet` clean on changed files; `swiftformat` applied to changed files (run selectively rather than repo-wide to avoid unrelated trailing-comma churn — pre-existing arrays were kept in the repo's no-trailing-comma style)
- [x] No breaking changes — additive only (new optional `init` param with default, new methods, new screen)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mDone/App/CalendarSelection.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    // MARK: - Hashable (identity by id only — CGColor isn't Hashable)
+
+    static func == (lhs: CalendarInfo, rhs: CalendarInfo) -> Bool {
+        lhs.id == rhs.id && lhs.title == rhs.title
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+    }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/App/CalendarSelection.swift]: Fixed in a8d0e18: `hash(into:)` now also combines `title` so `==` and `hash` are consistent, and the doc comment now accurately states identity is id + title (CGColor excluded as it isn't Hashable).

File: mDone/Services/CalendarService.swift [hunk 2]
```diff
@@ -16,11 +21,22 @@ actor CalendarService {
...
 
+    /// All event calendars the user has, for the selection UI. Also prunes
+    /// hidden entries for calendars that no longer exist.
+    func availableCalendars() -> [CalendarInfo] {
+        let calendars = eventStore.calendars(for: .event)
+        hiddenStore.prune(toExisting: Set(calendars.map(\.calendarIdentifier)))
+        return calendars
+            .map(CalendarInfo.init(from:))
+            .sorted { $0.title.localizedCaseInsensitiveCompare($1.title) == .orderedAscending }
+    }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Services/CalendarService.swift]: Fixed in a8d0e18: `availableCalendars()` is now a pure read. Pruning of stale hidden ids is an explicit step the caller performs (the selection screen calls `store.prune(toExisting:)` when it opens).

File: mDone/Views/Settings/CalendarSelectionView.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    Button(allShown ? "Hide All" : "Show All") {
+                        toggleAll()
+                    }
+                }
+            }
+            .task {
+                calendars = await appState.availableCalendars()
+                hidden = store.hiddenIdentifiers
+                isLoading = false
+            }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Settings/CalendarSelectionView.swift:69]: Fixed in a8d0e18: the screen now prunes the store explicitly and re-reads `store.hiddenIdentifiers` afterwards in `.task`, so the toggles reflect exactly what's persisted (including after prune drops stale ids).

File: mDone/Views/Calendar/CalendarScreen.swift [hunk 4]
```diff
@@ -55,6 +55,10 @@ struct CalendarScreen: View {
         .task(id: displayedMonth) {
             monthCalendarEvents = await appState.calendarEventsForMonth(displayedMonth)
         }
+        .task(id: appState.calendarFilterToken) {
+            dayCalendarEvents = await appState.calendarEventsForDate(selectedDate)
+            monthCalendarEvents = await appState.calendarEventsForMonth(displayedMonth)
+        }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Calendar/CalendarScreen.swift]: Fixed in a8d0e18: replaced the third `.task` with a `CalendarFetchKey(date:token:)` composite id folded into the existing day and month tasks. Each fires exactly once on first appear (no duplicate EventKit reads) and still refetches when the selection changes.
